overlord/snapstate/check_snap.go

```diff
@@ -176,7 +176,8 @@ var featureSet = map[string]bool{
 // https://docs.docker.com/engine/security/userns-remap/
 // https://github.com/lxc/lxd/blob/master/doc/userns-idmap.md
 var supportedSystemUsernames = map[string]uint32{
-	"snap_daemon": 584788,
+	"snap_daemon":   584788,
+	"snap_microk8s": 584789,
```

Comment:
if the snap is not installed with dangerous we really need to check for the snap-id to allow for the use of this user, we cannot just have the store-side check.

Reply:
Why is the store side check not enough? If I try to upload my foo snap that uses snap_microk8s system-username, it gets rejected the same way that uploading a snap that tries to plug k8s-support interface, etc

Basically, why is snap_microk8s more dangerous that k8s-support ?

Reply:
why do you say more?  is about the same, k8s-support is superprivileged so you need either dangerous or a snap-declaration. We could make this dependent on an interface instead of snapd-id but that adds quite  a bit complexity, because the code here is too early to check for what is connected.

Reply:
We don't have the same snap-id restriction on k8s-support that you are proposing for snap_microk8s, unless you mean to have the equivalent of `allow-installation: false` for system-usernames as well

Reply:
that's right, but notwithstanding what I might be hinted at, we have a namespace management issue here, and whatever the final mechanism via the store I don't expect we would give obviously snap_microk8s to everything with k8s-support, unless we think what we want is really snap_k8s ? or snap_kubernetes group? even then as I said because of namespace mgmt I would expect fewer snaps (only the one?) to own the group vs the interface

Reply:
That is a fair question about whether we should actually make it `snap_k8s`, there is some overlap between the microk8s snap and other snaps (like the juju snap I think?), perhaps @mardy can inquire about that.

Reply:
And, for example, the `kubernetes_worker` snap (https://github.com/charmed-kubernetes/snap-kubernetes-worker).

Reply:
@joedborg does that snap need to use the same group as setup by juju or microk8s though? I was under the impression that snap was meant as an isolated worker node that you run somewhere and it registers itself with the control plane that is on a different machine

Reply:
There are no plans to have a user group with elevated permissions for any other kubernetes snaps. MicroK8s is special in this regard.